stdlib/public/core/String.swift

```diff
@@ -476,7 +476,7 @@ extension String {
   ///       - buffer: A buffer covering uninitialized memory with room for the
   ///           specified number of UTF-8 code units.
   @inline(__always)
-  internal init(
+  public init(
```

Comment:
This PR probably shouldn't be merged until this goes through SE.

Reply:
Indeed